crates/configuration/src/shared/types.rs

```diff
@@ -231,6 +271,32 @@ impl Serialize for AssetLocation {
     }
 }
 
+impl<'de> Deserialize<'de> for AssetLocation {
+    fn deserialize<D>(deserializer: D) -> Result<AssetLocation, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        struct AssetLocationVisitor;
+
+        impl<'de> de::Visitor<'de> for AssetLocationVisitor {
+            type Value = AssetLocation;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+                formatter.write_str("a string")
+            }
+
+            fn visit_str<E>(self, v: &str) -> Result<Self::Value, E>
+            where
+                E: de::Error,
+            {
+                Ok(AssetLocation::from(v))
+            }
+        }
+
+        deserializer.deserialize_any(AssetLocationVisitor)
+    }
+}
```

Comment:
Same as above, remove the AssetLocationVisitor + Visitor implem trait out of the Deserialize implem